src/components/ui/EditList.tsx

```diff
@@ -0,0 +1,218 @@
+'use client'
+
+import { useTypeSafeTranslations } from '@/hooks/useTypeSafeTranslations'
+import { mergeClasses } from '@/lib/merge-classes'
+import { TranslationKey } from '@/types/translations'
+import { useRef, useState } from 'react'
+import Modal from '../modals/Modal'
+import Btn from './Btn'
+import IconBtn from './IconBtn'
+import TextInput from './TextInput'
+
+interface EditListItem {
+  id: string
+  value: string
+  numBooks?: number
+}
+
+interface EditListProps {
+  items: EditListItem[]
+  onItemEditSaveClick: (item: EditListItem, newValue: string) => Promise<any>
+  onItemDeleteClick: (item: EditListItem) => Promise<any>
+  saveConfirmI18nKey: TranslationKey
+  deleteConfirmI18nKey: TranslationKey
+  libraryId?: string
+}
+
+export default function EditList({ items, onItemEditSaveClick, onItemDeleteClick, saveConfirmI18nKey, deleteConfirmI18nKey, libraryId }: EditListProps) {
+  const i18n = useTypeSafeTranslations()
+  const [itemList, setItemList] = useState(items)
+  const [editedItem, setEditedItem] = useState<EditListItem>({ id: '', value: '' })
+  const [newValue, setNewValue] = useState('')
+  const [isProcessingModalOpen, setIsProcessingModalOpen] = useState(false)
+  const [isProcessing, setIsProcessing] = useState(false)
+  const delRef = useRef<EditListItem>(null)
+  const [isDeleting, setIsDeleting] = useState(false)
+  const showNumBooks = items.some((item) => {
+    return item.numBooks && item.numBooks > 0
+  })
+
+  // Switch to edit mode for a row
+  const handleEditItemClick = (clickedItem: EditListItem) => {
+    setEditedItem(clickedItem)
+    setNewValue(clickedItem.value)
+  }
+
+  // Delete a row
+  const handleDeleteClick = (clickedItem: EditListItem) => {
+    delRef.current = clickedItem
+    setIsDeleting(true)
+    setIsProcessingModalOpen(true)
+  }
+
+  const handleDeleteModalClick = async () => {
+    if (!delRef.current) return
+    setIsProcessing(true)
+    await onItemDeleteClick(delRef.current)
+      .then(() => {
```

Comment:
Use of .then with await is redundant. Also there's no error handling.
```suggestion
    try {
        await onItemDeleteClick(delref.current)
    } catch(e) {
        ...
    } finally {
        ...
    }
```